src/Compilation.zig

```diff
@@ -400,12 +412,39 @@ fn generateBuiltinTypes(comp: *Compilation) !void {
 
     const va_list = try comp.generateVaListType();
 
+    const pid_t: Type = switch (os) {
+        .haiku => .{ .specifier = .long },
+        // Todo: pid_t is required to "a signed integer type"; are there any systems
```

Comment:
This sounds like a 16 bit Kernel https://de.wikipedia.org/wiki/Embeddable_Linux_Kernel_Subset, where `sizeof(int)=16`. See also https://stackoverflow.com/questions/11438794/is-the-size-of-c-int-2-bytes-or-4-bytes.

Reply:
If `int` is 16 bits, then `short` will also be 16 bits since that's the minimum size and it can't be larger than `int`. So I think `short` and `int` would be equivalent in that case; the question is whether there are 32 bit systems with 16 bit pid_t since technically that seems to be allowed.

Reply:
After searching for a while I did not find any system, who would do that.
However, there is https://stackoverflow.com/questions/7829058/how-to-run-16-bit-code-on-32-bit-linux, for example via https://github.com/lkundrak/dev86/tree/master/elksemu or vm86 (`man vm86`) or qemu /simulation.

Sounds very niche though for (very) old software.